feat(bdpimport): Allow user to search for BDP projects instead of loading them all

bodetc opened the pull request:
Must be considered with PR https://github.com/sw360/sw360bdpImportService/pull/4 from bdpImportService

mcjaeger commented:
maybe, please consider to clean up the commit history.

bodetc commented:
commit history is changed

mcjaeger commented:
in order to test the mock functionality is "mock" for the server value

maierthomas commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortalConstants.java:
```diff
@@ -330,9 +330,9 @@
-    public static final String SESSION_IMPORT_DB = "importProject_import_db";
     public static final String SESSION_IMPORT_PASS = "importProject_import_pass";
     public static final String SESSION_IMPORT_URL = "importProject_import_url";
+    public static final String IMPORT_PROJECT_NAME = "importProject_import_project_name";
```
Is IMPORT_PROJECT_NAME really a import credential or is this constant more like a key which help to specify the the result of the import data?

maierthomas replied:
Its only a small recommendation but the sections Import Import Projects, Import credentials, Import Actions, Import Status are not very clearly formatted and listed

bodetc replied:
I change the variable names and string content in this section to make it more coherent. Actually, I was wondering if it would make sense to move all those constants to a new class. Or we have some guidelines about having them in one place?

maierthomas replied:
I dont know if we have any guidelines for the naming here but the updated version looks much better!

maierthomas commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectImportPortlet.java:
```diff
@@ -85,7 +87,7 @@ private RemoteCredentials getRemoteCredentialsFromSession(PortletSession session
     }
 
-    private void putRemoteCredentialsIntoSession(PortletSession session, RemoteCredentials remoteCredentials) {
+    private void putRemoteCredentialsAndProjectNameIntoSession(PortletSession session, RemoteCredentials remoteCredentials) {
```
Why did you rename `putRemoteCredentialsIntoSession`?
In this method I can not see any put command for the project name.

bodetc replied:
My bad, I reverted this.

maierthomas commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectImportPortlet.java:
```diff
@@ -98,8 +100,9 @@ public void doView(RenderRequest request, RenderResponse response) throws IOExce
+            importables = loadImportables(reCred, projectName);
             loggedIn = true;
             loggedInServer = reCred.getServerUrl();
         }
```
maybe we can use constants for the request.setAttribute calls
`idName, importables, loggedIn, loggedInServer`

maierthomas commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectImportPortlet.java:
```diff
@@ -252,23 +253,23 @@ void setNewImportSource(RemoteCredentials newCredentials, PortletSession session
-                if (p.isSetExternalIds() && ! isNullOrEmpty(p.getExternalIds().get(getIdName())))
-                jsonObject.put("externalId", p.getExternalIds().get(getIdName()));
+                if (p.isSetExternalIds() && !isNullOrEmpty(p.getExternalIds().get(getIdName())))
+                    jsonObject.put("externalId", p.getExternalIds().get(getIdName()));
```
maybe use constants for 
`name, externalId`

maierthomas commented on frontend/sw360-portlet/src/main/webapp/html/projects/import.jsp:
```diff
@@ -78,13 +79,23 @@
+        </div>
+    </div>
+    <input type="button" onclick="updateDataSource()" value="Connect" id="buttonConnect"/>
+    <input type="button" onclick="updateDataSource()" value="Refresh" class="hidden" id="buttonRefresh"/>
```
Is there a special reason why you added a new button instead of renaming the attribute value of the first button?

bodetc replied:
No, just that the prexisiting logic was already showing/hidding buttons instead of renaming them, so I kept using that.

maierthomas replied:
Ok

mcjaeger commented:
I am not sure, there seem to be errors on the page when entering bogus values, but not having loaded the bpdImport project:

<img width="1629" alt="screen shot 2017-11-16 at 03 24 44" src="https://user-images.githubusercontent.com/3418977/32870825-f98942ea-ca7d-11e7-94b5-5f675e8a4c8a.png">

mcjaeger commented:
maybe merge PR sw360/sw360bdpImportService#4 first and then ask users to test the PR with the bpdImportService

mcjaeger commented:
testing is good,looks good, but I am not sure to merge ... @bodetc do you plan to change names the `PROJECT_IMPORT_USER_ACTION__IMPORT_BDP` and the similar?

mcjaeger commented:
@bodetc I think there is some confusion at my side now.

I am not sure if you are going for the renaming thing as mentioned in #680? ... it seems to apply also here. My confusion is now, if I merge now, how will you change the naming of constant for #680? Maybe it would be good to consider the renaming of the constants in this PR?

bodetc commented:
@mcjaeger I have renamed the constants and removed "bdp" from Strings in this PR (pushed a few minutes ago).

For renaming the thrift file, I will do this in PR https://github.com/sw360/sw360portal/pull/680 as this would be a breaking change to the BDP Import Service as well, and I need to adapt the PR https://github.com/sw360/sw360bdpImportService/pull/5 there accordingly
